### PR TITLE
[Access-tokens] Update access-tokens to version v0.0.6

### PR DESCRIPTION
**Pull request by jprzychodzen**
Changelog:
* Add layer of parallelization per token
* Make requests cheaper with  ResoruceVersion : 0

/sig scalability

### CODE REVIEW

**Comment by mm4tt** [comment]
Please revert your changes in known-flags.txt, it's not needed to update it anymore, see https://github.com/kubernetes/perf-tests/pull/1006

**Comment by jkaniuk** [comment]
> Please revert your changes in known-flags.txt, it's not needed to update it anymore, see #1006

Why do we have it then?

**Comment by mm4tt** [comment]
@jkaniuk, we don't have it anymore, it was removed in https://github.com/kubernetes/perf-tests/pull/1006

File: util-images/access-tokens/cmd/main.go [hunk 1]
```diff
@@ -103,5 +109,6 @@ func newConfig(tokenFile, rootCAFile string) (*rest.Config, error) {
 		TLSClientConfig: tlsClientConfig,
 		BearerToken:     string(token),
 		BearerTokenFile: tokenFile,
+		RateLimiter:     flowcontrol.NewFakeAlwaysRateLimiter(),
```
Comment (mm4tt) [inline util-images/access-tokens/cmd/main.go]: Cannot you just set QPS: *qps here?

File: util-images/access-tokens/cmd/main.go [hunk 2]
```diff
@@ -29,18 +29,21 @@ import (
...
 )
 
 var (
-	namespace = flag.String("namespace", "", "Namespace that is targeted with request flow.")
-	tokenDirs = flag.StringSlice("access-token-dirs", nil, "Directories with access-token. A worker will be started per each dir/token")
-	qps       = flag.Float64("qps-per-worker", 0.5, "QPS per worker")
+	namespace       = flag.String("namespace", "", "Namespace that is targeted with request flow.")
+	tokenDirs       = flag.StringSlice("access-token-dirs", nil, "Directories with access-token. A worker will be started per each dir/token")
+	qps             = flag.Float64("qps-per-worker", 0.5, "QPS per worker")
+	workersPerToken = flag.Int("workers-per-token", 1, "Number of workers using the same token")
```
Comment (mm4tt) [inline util-images/access-tokens/cmd/main.go]: Why do we need this additional layer here of parallelism here? Is there any use-case for it?

The downside of it is that it's getting more complicated, total qps is now: #tokens * qps * worksersPerToken

File: util-images/access-tokens/cmd/main.go (hunk 1 above)
Comment (jprzychodzen) [reply util-images/access-tokens/cmd/main.go]: It's more complicated than that:
  1. We are already rate limiting in `func worker` - we do not need another limiter here
  2. Setting a QPS would require to set Burst parameter. It raises a question what would be sane value for Burst.

Comment (mm4tt) [reply util-images/access-tokens/cmd/main.go]: Setting burst to 1 should work, we need 1 initial request and then the the rate limiter's bucket should be refilled with the exact qps we use.

But I see your point and I agree. There is no point in having two rate-limiting mechanism.

Could you just add a comment here explaining that, basically what you wrote in point 1. ?

Comment (jprzychodzen) [reply util-images/access-tokens/cmd/main.go]: Added, done.

File: util-images/access-tokens/cmd/main.go (hunk 2 above)
Comment (jprzychodzen) [reply util-images/access-tokens/cmd/main.go]: It was mostly related to broken implementation of `func worker` that could be throttled in a case if server in not responding fast enough. Changed implementation to asynchronous and removed `workersPerToken` flag

**Pull request merged**